src/clarity_vm/clarity.rs

```diff
@@ -903,6 +800,9 @@ impl<'a> ClarityBlockConnection<'a> {
                     })
                     .unwrap();
 
+                // require 2.1 rules henceforth in this connection as well
+                tx_conn.epoch = StacksEpochId::Epoch21;
```

Comment:
This will only set the transaction's epoch, _not_ the block's epoch. `ClarityBlockConnection ::start_transaction_processing` is the common method used to start a transaction, and it sets the `epoch` to the block connection's epoch. So, I think that you need to update `self.epoch`.

Reply:
Sure, will update both.

Reply:
- [ ]